presto-geospatial/src/test/java/io/prestosql/plugin/geospatial/TestGeoFunctions.java

```diff
@@ -472,6 +472,70 @@ public void testLineLocatePoint()
         assertInvalidFunction("line_locate_point(ST_GeometryFromText('LINESTRING (0 0, 0 1, 2 1)'), ST_GeometryFromText('POLYGON ((1 1, 1 4, 4 4, 4 1))'))", "Second argument to line_locate_point must be a Point. Got: Polygon");
     }
 
+    @Test
+    public void testLineInterpolatePoint()
+    {
+        assertFunction("ST_AsText(line_interpolate_point(ST_GeometryFromText('LINESTRING EMPTY'), 0.5))", VARCHAR, null);
+
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1, 10 10)", 0.0, 0.0, 0.0);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1, 10 10)", 0.1, 1.0, 1.0);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1, 10 10)", 0.05, 0.5, 0.5);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1, 10 10)", 0.4, 4.000000000000001, 4.000000000000001);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1, 10 10)", 01.0, 10.0, 10.0);
+
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1)", 0.0, 0.0, 0.0);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1)", 0.1, 0.1, 0.1);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1)", 0.05, 0.05, 0.05);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1)", 0.4, 0.4, 0.4);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 1)", 1.0, 1.0, 1.0);
+
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 0, 1 9)", 0.0, 0.0, 0.0);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 0, 1 9)", 0.1, 1.0, 0.0);
+        assertLineInterpolatePoint("LINESTRING (0 0, 1 0, 1 9)", 0.5, 1.0, 4.0);
```

Comment:
Can you add a test where the interpolated point is in the second segment?

Reply:
Sorry, I'm not too sure what you mean. In the test case on line 494, the interpolated point is on the (1 0, 1 9) segment at (1 4). I did add a test case though that falls on the (0 0, 1 0) segment though.

Reply:
You’re right! I misread the test. Disregard this comment.